lib/bibliothecary/multi_parsers/target_framework.rb

```diff
@@ -0,0 +1,43 @@
+module Bibliothecary
+  module MultiParsers
+    module TargetFramework
+      TARGET_FRAMEWORKS = {
+        '.NET': ['net11', 'net20', 'net35', 'net40', 'net403', 'net45', 'net451', 'net452', 'net46', 'net461', 'net462', 'net47', 'net471', 'net472', 'net48'],
+        '.NET Core': ['netcoreapp1.0', 'netcoreapp1.1', 'netcoreapp2.0', 'netcoreapp2.1', 'netcoreapp2.2', 'netcoreapp3.0', 'netcoreapp3.1', 'net5.0', 'net6.0', 'net7.0']
+      }.freeze
+
+      def identify_target_framework(tfm)
+        matching_framework = find_matching_framework(tfm)
+        return if matching_framework.nil?
+
+        version = extract_version(tfm)
+        {
+          name: matching_framework.to_s,
+          requirement: version,
+          type: 'runtime'
+        }
+      rescue
+        {}
+      end
+
+      def dotnet_framework_version(tfm)
+        tfm.scan(/\d/).join('.')
+      rescue
```

Comment:
remove rescue. We need to know if there are any errors. It will show up on sentry. If it shows up we can fix it again.

Reply:
Hi @AkBacancy if there is no rescue  here, and if there any error happens according to existing gem code https://github.com/stackshareio/bibliothecary/blob/main/lib/bibliothecary/parsers/nuget.rb, it won't raise error instead it will return empty array. So How i designed is, if there is an error happened while fetching version, return nil, so the gem will return tool, but without version. If there is no rescue gem will not output tool too.